### PR TITLE
Single container db alter

### PR DESCRIPTION
**Pull request by mkopsnc**
Lots of changes inside single-container.

I must admid, I was fed up with long builds related to all the downloads of Maven dependencies.

Now, it's possible to pass `catalog_qt_2` with already compiled code. In that case, Dockerfile will not build everything from the scratch.

Alternative approach is to create second layer with only pom.xml files and download dependencies (we may think about that as well).

For now, what we have, will do. I have also moved all the mysql related scripts into separate files. Dockerfile started to become huge and it was hard to follow all the steps inside. Now, you have to reach to external file, but at least we don't have long chains of execution.

### CODE REVIEW

**Comment by mkopsnc** [comment]
@tzok, @agatafilipczak - if you have any comments, let me know before I merge this one.

**Comment by tzok** [comment]
IMHO, it would be beneficial to use a staged building process in Dockerfile. One stage will build the Java projects. Others will just copy the resulting binaries (i.e. JARs). The rest is left to Docker caching mechanisms. This way, the Java project is compiled only once per introduced changes. Maybe in the future, we can think of a solution like that.

File: single-container/Dockerfile [hunk 1]
```diff
@@ -83,9 +82,13 @@ RUN tar xf catalog_qt_2.tar
...
-COPY --chown=imas scripts/compile-java-code.sh /home/imas/opt/compile-java-code.sh
-RUN chmod +x /home/imas/opt/compile-java-code.sh
-RUN /home/imas/opt/compile-java-code.sh
+COPY --chown=imas scripts/compile-scripts/* /home/imas/opt/
+RUN chmod +x /home/imas/opt/compile-*.sh
+
+RUN /home/imas/opt/compile-common.sh /home/imas/opt/catalog_qt_2
+RUN /home/imas/opt/compile-plugin.sh /home/imas/opt/catalog_qt_2
+RUN /home/imas/opt/compile-client.sh /home/imas/opt/catalog_qt_2
+RUN /home/imas/opt/compile-server.sh /home/imas/opt/catalog_qt_2
```
Comment (tzok) [inline single-container/Dockerfile:91]: Usually, subsequent `RUN` lines are combined with the `&&` operator. This makes the resulting image smaller.

**Review by tzok — APPROVED** [review]

File: single-container/Dockerfile (hunk 1 above)
Comment (mkopsnc) [reply single-container/Dockerfile:91]: OK - I will change that to

```
RUN /home/imas/opt/compile-common.sh /home/imas/opt/catalog_qt_2 && \
/home/imas/opt/compile-plugin.sh /home/imas/opt/catalog_qt_2 && \
/home/imas/opt/compile-client.sh /home/imas/opt/catalog_qt_2 && \
/home/imas/opt/compile-server.sh /home/imas/opt/catalog_qt_2
```

Comment (mkopsnc) [reply single-container/Dockerfile:91]: > IMHO, it would be beneficial to use a staged building process in Dockerfile. One stage will build the Java projects. Others will just copy the resulting binaries (i.e. JARs). The rest is left to Docker caching mechanisms. This way, the Java project is compiled only once per introduced changes. Maybe in the future, we can think of a solution like that.

Yes, exactly. I think we definitely should move towards this approach, but I have to make some research here. The major issue here is not the build process itself - it takes just a minute or so. The biggest problem is related to `maven` downloading all the dependencies each time we run `build` command.

There is a way to overcome this issue. We can extract only `pom.xml` files, download dependencies in one of the early stages, and then build on top of that. This way, we will avoid downloading dependencies as long as `pom.xml` files stay the same. However, there is one issue. We have to change the way of building a little bit.

Now, `catalog_qt_2.tar` is loaded into image as one of stages. So, we are already lost because it will trigger all the subsequent stages. This is why instead of passing `tar` file I have to pass directory, and before copying directory into image, I have to copy just a `pom.xml` files in a first place. This is doable but requires little bit of the effort to prepare and test the whole solution.
